modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/AbstractProcessor.java

```diff
@@ -16,10 +16,12 @@
 abstract class AbstractProcessor implements Processor {
     private final String tag;
     private final String description;
+    private final Boolean ignoreFailure;
 
-    protected AbstractProcessor(String tag, String description) {
+    protected AbstractProcessor(String tag, String description, Boolean ignoreFailure) {
         this.tag = tag;
         this.description = description;
+        this.ignoreFailure = ignoreFailure;
     }
```

Comment:
We should move `AbstractProcessor` into core, next to the `Processor` interface, so that we don't need to duplicate these flags.

Also, `ignoreFailure` should be lower-cased `boolean`, because it should not be nullable.